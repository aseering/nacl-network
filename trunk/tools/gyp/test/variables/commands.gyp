# This is a simple test file to make sure that variable substitution
# happens correctly.  Run "run_tests.py" using python to generate the
# output from this gyp file.

{
  'variables': {
    'pi': 'import math; print math.pi',
    'third_letters': "<(other_letters)HIJK",
    'letters_list': 'ABCD',
    'other_letters': '<(letters_list)EFG',
    'check_included': '<(included_variable)',
    'check_lists': [
      '<(included_variable)',
      '<(third_letters)',
    ],
  },
  'includes': [
    'commands.gypi',
  ],
  'targets': [
    {
      'target_name': 'foo',
      'type': 'none',
      'variables': {
        'var1': '<!(["python", "-c", "<(pi)"])',
        'var2': '<!(python -c "print \'<!(python -c "<(pi)") <(letters_list)\'")',
        'var3': '<!(python -c "print \'<(letters_list)\'")',
        'var4': '<(<!(python -c "print \'letters_list\'"))',
        'var5': 'letters_',
        'var6': 'list',
      },
      'actions': [
        {
          'action_name': 'test_action',
          'variables': {
            'var7': '<!(echo <(var5)<(var6))',
          },
          'inputs' : [
            '<(var2)',
          ],
          'outputs': [
            '<(var4)',
            '<(var7)',
          ],
          'action': [
            'echo',
            '<(_inputs)',
            '<(_outputs)',
          ],
        },
      ],
    },
  ],
}
