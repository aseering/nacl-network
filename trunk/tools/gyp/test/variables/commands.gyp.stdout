GENERAL: running with these options:
GENERAL:   msvs_version: None
GENERAL:   suffix: ''
GENERAL:   includes: None
GENERAL:   depth: '.'
GENERAL:   generator_flags: []
GENERAL:   generator_output: None
GENERAL:   formats: ['gypd']
GENERAL:   debug: ['variables', 'general']
GENERAL:   defines: None
VARIABLES: Matches: {'content': 'letters_list', 'is_array': '', 'type': '<', 'replace': '<(letters_list)'}
VARIABLES: Expanding 'letters_list' to 'letters_list'
VARIABLES: Found output 'ABCDEFG', recursing.
VARIABLES: Expanding 'ABCDEFG' to 'ABCDEFG'
VARIABLES: Expanding 'ABCDEFG' to 'ABCDEFG'
VARIABLES: Expanding 'XYZ' to 'XYZ'
VARIABLES: Matches: {'content': 'other_letters', 'is_array': '', 'type': '<', 'replace': '<(other_letters)'}
VARIABLES: Expanding 'other_letters' to 'other_letters'
VARIABLES: Found output '<(letters_list)EFGHIJK', recursing.
VARIABLES: Matches: {'content': 'letters_list', 'is_array': '', 'type': '<', 'replace': '<(letters_list)'}
VARIABLES: Expanding 'letters_list' to 'letters_list'
VARIABLES: Found output 'ABCDEFGHIJK', recursing.
VARIABLES: Expanding 'ABCDEFGHIJK' to 'ABCDEFGHIJK'
VARIABLES: Expanding 'ABCDEFGHIJK' to 'ABCDEFGHIJK'
VARIABLES: Expanding '<(letters_list)EFGHIJK' to 'ABCDEFGHIJK'
VARIABLES: Expanding 'ABCD' to 'ABCD'
VARIABLES: Expanding 'import math; print math.pi' to 'import math; print math.pi'
VARIABLES: Matches: {'content': 'included_variable', 'is_array': '', 'type': '<', 'replace': '<(included_variable)'}
VARIABLES: Expanding 'included_variable' to 'included_variable'
VARIABLES: Found output 'XYZ', recursing.
VARIABLES: Expanding 'XYZ' to 'XYZ'
VARIABLES: Expanding 'XYZ' to 'XYZ'
VARIABLES: Matches: {'content': 'included_variable', 'is_array': '', 'type': '<', 'replace': '<(included_variable)'}
VARIABLES: Expanding 'included_variable' to 'included_variable'
VARIABLES: Found output 'XYZ', recursing.
VARIABLES: Expanding 'XYZ' to 'XYZ'
VARIABLES: Expanding 'XYZ' to 'XYZ'
VARIABLES: Matches: {'content': 'third_letters', 'is_array': '', 'type': '<', 'replace': '<(third_letters)'}
VARIABLES: Expanding 'third_letters' to 'third_letters'
VARIABLES: Found output '<(other_letters)HIJK', recursing.
VARIABLES: Matches: {'content': 'other_letters', 'is_array': '', 'type': '<', 'replace': '<(other_letters)'}
VARIABLES: Expanding 'other_letters' to 'other_letters'
VARIABLES: Found output '<(letters_list)EFGHIJK', recursing.
VARIABLES: Matches: {'content': 'letters_list', 'is_array': '', 'type': '<', 'replace': '<(letters_list)'}
VARIABLES: Expanding 'letters_list' to 'letters_list'
VARIABLES: Found output 'ABCDEFGHIJK', recursing.
VARIABLES: Expanding 'ABCDEFGHIJK' to 'ABCDEFGHIJK'
VARIABLES: Expanding 'ABCDEFGHIJK' to 'ABCDEFGHIJK'
VARIABLES: Expanding '<(letters_list)EFGHIJK' to 'ABCDEFGHIJK'
VARIABLES: Expanding '<(other_letters)HIJK' to 'ABCDEFGHIJK'
VARIABLES: Expanding '.' to '.'
VARIABLES: Expanding 'letters_' to 'letters_'
VARIABLES: Matches: {'content': '<!(python -c "print \'letters_list\'"', 'is_array': '', 'type': '<', 'replace': '<(<!(python -c "print \'letters_list\'")'}
VARIABLES: Matches: {'content': 'python -c "print \'letters_list\'"', 'is_array': '', 'type': '<!', 'replace': '<!(python -c "print \'letters_list\'")'}
VARIABLES: Expanding 'python -c "print 'letters_list'"' to 'python -c "print 'letters_list'"'
VARIABLES: Executing command 'python -c "print 'letters_list'"'
VARIABLES: Found output 'letters_list', recursing.
VARIABLES: Expanding 'letters_list' to 'letters_list'
VARIABLES: Expanding 'letters_list' to 'letters_list'
VARIABLES: Found output 'ABCD', recursing.
VARIABLES: Expanding 'ABCD' to 'ABCD'
VARIABLES: Expanding 'ABCD' to 'ABCD'
VARIABLES: Expanding 'list' to 'list'
VARIABLES: Matches: {'content': '"python", "-c", "<(pi', 'is_array': '[', 'type': '<!', 'replace': '<!(["python", "-c", "<(pi)'}
VARIABLES: Matches: {'content': 'pi', 'is_array': '', 'type': '<', 'replace': '<(pi)'}
VARIABLES: Expanding 'pi' to 'pi'
VARIABLES: Found output '["python", "-c", "import math; print math.pi"]', recursing.
VARIABLES: Expanding '["python", "-c", "import math; print math.pi"]' to '["python", "-c", "import math; print math.pi"]'
VARIABLES: Expanding '["python", "-c", "import math; print math.pi"]' to '["python", "-c", "import math; print math.pi"]'
VARIABLES: Executing command '['python', '-c', 'import math; print math.pi']'
VARIABLES: Found output '3.14159265359', recursing.
VARIABLES: Expanding '3.14159265359' to '3.14159265359'
VARIABLES: Expanding '3.14159265359' to '3.14159265359'
VARIABLES: Matches: {'content': 'python -c "print \'<(letters_list', 'is_array': '', 'type': '<!', 'replace': '<!(python -c "print \'<(letters_list)'}
VARIABLES: Matches: {'content': 'letters_list', 'is_array': '', 'type': '<', 'replace': '<(letters_list)'}
VARIABLES: Expanding 'letters_list' to 'letters_list'
VARIABLES: Found output 'python -c "print 'ABCD'"', recursing.
VARIABLES: Expanding 'python -c "print 'ABCD'"' to 'python -c "print 'ABCD'"'
VARIABLES: Expanding 'python -c "print 'ABCD'"' to 'python -c "print 'ABCD'"'
VARIABLES: Executing command 'python -c "print 'ABCD'"'
VARIABLES: Found output 'ABCD', recursing.
VARIABLES: Expanding 'ABCD' to 'ABCD'
VARIABLES: Expanding 'ABCD' to 'ABCD'
VARIABLES: Matches: {'content': 'letters_list', 'is_array': '', 'type': '<', 'replace': '<(letters_list)'}
VARIABLES: Expanding 'letters_list' to 'letters_list'
VARIABLES: Matches: {'content': 'python -c "print \'<!(python -c "<(pi', 'is_array': '', 'type': '<!', 'replace': '<!(python -c "print \'<!(python -c "<(pi)'}
VARIABLES: Matches: {'content': 'python -c "<(pi', 'is_array': '', 'type': '<!', 'replace': '<!(python -c "<(pi)'}
VARIABLES: Matches: {'content': 'pi', 'is_array': '', 'type': '<', 'replace': '<(pi)'}
VARIABLES: Expanding 'pi' to 'pi'
VARIABLES: Found output 'python -c "import math; print math.pi"', recursing.
VARIABLES: Expanding 'python -c "import math; print math.pi"' to 'python -c "import math; print math.pi"'
VARIABLES: Expanding 'python -c "import math; print math.pi"' to 'python -c "import math; print math.pi"'
VARIABLES: Executing command 'python -c "import math; print math.pi"'
VARIABLES: Found output 'python -c "print '3.14159265359 ABCD'"', recursing.
VARIABLES: Expanding 'python -c "print '3.14159265359 ABCD'"' to 'python -c "print '3.14159265359 ABCD'"'
VARIABLES: Expanding 'python -c "print '3.14159265359 ABCD'"' to 'python -c "print '3.14159265359 ABCD'"'
VARIABLES: Executing command 'python -c "print '3.14159265359 ABCD'"'
VARIABLES: Found output '3.14159265359 ABCD', recursing.
VARIABLES: Expanding '3.14159265359 ABCD' to '3.14159265359 ABCD'
VARIABLES: Expanding '3.14159265359 ABCD' to '3.14159265359 ABCD'
VARIABLES: Expanding 'foo' to 'foo'
VARIABLES: Expanding 'none' to 'none'
VARIABLES: Matches: {'content': 'var6', 'is_array': '', 'type': '<', 'replace': '<(var6)'}
VARIABLES: Expanding 'var6' to 'var6'
VARIABLES: Matches: {'content': 'echo <(var5', 'is_array': '', 'type': '<!', 'replace': '<!(echo <(var5)'}
VARIABLES: Matches: {'content': 'var5', 'is_array': '', 'type': '<', 'replace': '<(var5)'}
VARIABLES: Expanding 'var5' to 'var5'
VARIABLES: Found output 'echo letters_list', recursing.
VARIABLES: Expanding 'echo letters_list' to 'echo letters_list'
VARIABLES: Expanding 'echo letters_list' to 'echo letters_list'
VARIABLES: Executing command 'echo letters_list'
VARIABLES: Found output 'letters_list', recursing.
VARIABLES: Expanding 'letters_list' to 'letters_list'
VARIABLES: Expanding 'letters_list' to 'letters_list'
VARIABLES: Expanding 'test_action' to 'test_action'
VARIABLES: Expanding 'echo' to 'echo'
VARIABLES: Matches: {'content': '_inputs', 'is_array': '', 'type': '<', 'replace': '<(_inputs)'}
VARIABLES: Expanding '_inputs' to '_inputs'
VARIABLES: Matches: {'content': 'var2', 'is_array': '', 'type': '<', 'replace': '<(var2)'}
VARIABLES: Expanding 'var2' to 'var2'
VARIABLES: Found output '3.14159265359 ABCD', recursing.
VARIABLES: Expanding '3.14159265359 ABCD' to '3.14159265359 ABCD'
VARIABLES: Expanding '3.14159265359 ABCD' to '3.14159265359 ABCD'
VARIABLES: Found output '"3.14159265359 ABCD"', recursing.
VARIABLES: Expanding '"3.14159265359 ABCD"' to '"3.14159265359 ABCD"'
VARIABLES: Expanding '"3.14159265359 ABCD"' to '"3.14159265359 ABCD"'
VARIABLES: Matches: {'content': '_outputs', 'is_array': '', 'type': '<', 'replace': '<(_outputs)'}
VARIABLES: Expanding '_outputs' to '_outputs'
VARIABLES: Matches: {'content': 'var4', 'is_array': '', 'type': '<', 'replace': '<(var4)'}
VARIABLES: Expanding 'var4' to 'var4'
VARIABLES: Found output 'ABCD', recursing.
VARIABLES: Expanding 'ABCD' to 'ABCD'
VARIABLES: Expanding 'ABCD' to 'ABCD'
VARIABLES: Matches: {'content': 'var7', 'is_array': '', 'type': '<', 'replace': '<(var7)'}
VARIABLES: Expanding 'var7' to 'var7'
VARIABLES: Found output 'letters_list', recursing.
VARIABLES: Expanding 'letters_list' to 'letters_list'
VARIABLES: Expanding 'letters_list' to 'letters_list'
VARIABLES: Found output 'ABCD letters_list', recursing.
VARIABLES: Expanding 'ABCD letters_list' to 'ABCD letters_list'
VARIABLES: Expanding 'ABCD letters_list' to 'ABCD letters_list'
VARIABLES: Expanding '3.14159265359 ABCD' to '3.14159265359 ABCD'
VARIABLES: Expanding 'ABCD' to 'ABCD'
VARIABLES: Expanding 'letters_list' to 'letters_list'
VARIABLES: Expanding 'dummy' to 'dummy'
VARIABLES: Expanding 'none' to 'none'
VARIABLES: Expanding 'commands.gyp' to 'commands.gyp'
VARIABLES: Expanding 'commands.gypi' to 'commands.gypi'
VARIABLES: Expanding 'dummy' to 'dummy'
VARIABLES: Expanding 'none' to 'none'
VARIABLES: Expanding 'dummy' to 'dummy'
VARIABLES: Expanding 'none' to 'none'
VARIABLES: Expanding 'letters_' to 'letters_'
VARIABLES: Expanding 'ABCD' to 'ABCD'
VARIABLES: Expanding 'list' to 'list'
VARIABLES: Expanding '3.14159265359' to '3.14159265359'
VARIABLES: Expanding 'ABCD' to 'ABCD'
VARIABLES: Expanding '3.14159265359 ABCD' to '3.14159265359 ABCD'
VARIABLES: Expanding 'foo' to 'foo'
VARIABLES: Expanding 'none' to 'none'
VARIABLES: Expanding 'letters_list' to 'letters_list'
VARIABLES: Expanding 'test_action' to 'test_action'
VARIABLES: Expanding 'echo' to 'echo'
VARIABLES: Expanding '"3.14159265359 ABCD"' to '"3.14159265359 ABCD"'
VARIABLES: Expanding 'ABCD letters_list' to 'ABCD letters_list'
VARIABLES: Expanding '3.14159265359 ABCD' to '3.14159265359 ABCD'
VARIABLES: Expanding 'ABCD' to 'ABCD'
VARIABLES: Expanding 'letters_list' to 'letters_list'
